pkg/operator/reports.go

```diff
@@ -344,7 +344,15 @@ func getReportPeriod(now time.Time, logger log.FieldLogger, report *metering.Rep
 // according the report's schedule. If the next scheduled reporting period
 // hasn't elapsed, runReport will requeue the resource for a time when
 // the period has elapsed.
+// Report expiration is checked before running, report is deleted if expired.
 func (op *Reporting) runReport(logger log.FieldLogger, report *metering.Report) error {
+	// check if the report is expired
+	if reportExpired := isReportExpired(logger, report, time.Now()); reportExpired {
+		deleteReport(report, op)
+		return nil
+	} else {
```

Comment:
Hmm, I wonder if we should instead just set the Report's `metadata.DeletionTimestamp` instead, and leaving the deletion of this object to the updateFunc for the Report informer.

Reply:
I don't think that's how custom resources get deleted. But then again calling `op.deleteReport` isn't deleting them AND doesn't seem to be setting the `DeletionTimestamp`, so what do I know.